### ISSUE TITLE
Ex 07 erroring during run

### ISSUE THREAD

**Issue by architkulkarni** [dmatrix/ray-core-serve-tutorial-mlops#20]
Following the instructions:
```
conda create -n ray-core-serve-tutorial python=3.8
conda activate ray-core-serve-tutorial
git clone git@github.com:dmatrix/ray-core-serve-tutorial-mlops.git
cd to <cloned_dir>
python3 -m pip install -r requirements.txt
python3 -m ipykernel install
conda install jupyterlab
jupyter lab
```
and running all cells in Ex 07, I get the following two errors, and I'm not sure if they're related:
Running step 4
```
run_id = create_and_save_model()
# Construct model uri to load the model from our model registry
run_id = create_and_save_model()
# Construct model uri to load the model from our model registry
uri = f"runs:/{run_id}/model"
```
I get
```
2022/06/03 13:47:16 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during sklearn autologging: No module named 'matplotlib'
```
But I have matplotlib installed in the conda env, so I'm not sure what's going on.

Step 5 passes fine, but step 6
```
# Deploy our model.
BoostingModel.deploy(uri)
```
results in 
```python-traceback
2022-06-03 13:47:21,801	INFO api.py:615 -- Updating deployment 'BoostingModel'. component=serve deployment=BoostingModel
(ServeController pid=22606) 2022-06-03 13:47:21,892	INFO deployment_state.py:1216 -- Adding 1 replicas to deployment 'BoostingModel'. component=serve deployment=BoostingModel
(ServeController pid=22606) 2022-06-03 13:47:23,468	ERROR deployment_state.py:421 -- Exception in deployment 'BoostingModel'
(ServeController pid=22606) Traceback (most recent call last):
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/ray/serve/deployment_state.py", line 412, in check_ready
(ServeController pid=22606)     deployment_config, version = ray.get(self._ready_obj_ref)
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
(ServeController pid=22606)     return func(*args, **kwargs)
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/ray/worker.py", line 1809, in get
(ServeController pid=22606)     raise value.as_instanceof_cause()
(ServeController pid=22606) ray.exceptions.RayTaskError(OSError): ray::RayServeWrappedReplica.reconfigure() (pid=22616, ip=127.0.0.1, repr=<ray.serve.replica.BoostingModel object at 0x7fb4b78b9e50>)
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/concurrent/futures/_base.py", line 437, in result
(ServeController pid=22606)     return self.__get_result()
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
(ServeController pid=22606)     raise self._exception
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/ray/serve/replica.py", line 200, in reconfigure
(ServeController pid=22606)     await self._initialize_replica()
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/ray/serve/replica.py", line 141, in initialize_replica
(ServeController pid=22606)     await sync_to_async(_callable.__init__)(*init_args, **init_kwargs)
(ServeController pid=22606)   File "/var/folders/75/7kpb8gj57k11df4ksj9lnt900000gn/T/ipykernel_22503/2862471892.py", line 6, in __init__
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/mlflow/pyfunc/__init__.py", line 716, in load_model
(ServeController pid=22606)     local_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/mlflow/tracking/artifact_utils.py", line 94, in _download_artifact_from_uri
(ServeController pid=22606)     return get_artifact_repository(artifact_uri=root_uri).download_artifacts(
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/mlflow/store/artifact/runs_artifact_repo.py", line 125, in download_artifacts
(ServeController pid=22606)     return self.repo.download_artifacts(artifact_path, dst_path)
(ServeController pid=22606)   File "/Users/archit/anaconda3/envs/ray-core-serve-tutorial/lib/python3.8/site-packages/mlflow/store/artifact/local_artifact_repo.py", line 79, in download_artifacts
(ServeController pid=22606)     raise IOError("No such file or directory: '{}'".format(local_artifact_path))
```

**Comment by dmatrix** [comment]
I fixed the problem. Included matplotlib in the requirements.txt 
and imported into the nb.

![Screen Shot 2022-06-03 at 9 55 24 PM](https://user-images.githubusercontent.com/1117597/171983857-bb6604ce-65cd-4287-a461-1aae3e222940.png)

![Screen Shot 2022-06-03 at 9 56 48 PM](https://user-images.githubusercontent.com/1117597/171983912-41690700-52a5-4273-ae50-2b60027b9491.png)

Will file PR and merge it.

**Comment by dmatrix** [comment]
Fixed in PR #23
